client/src/components/Gallery/Gallery.js

```diff
@@ -58,109 +104,60 @@ export default class Gallery extends React.Component {
         )
     }
 
-    renderSecondaryImages(images) {
-        console.log(images.length);
-        let extraImages = [];
-        if (images.length > 5) {
-            extraImages = images.slice(4);
-            images = images.slice(0, 4);
-        }
-        let count = images.length;
-        if (extraImages.length) count++;
-        const imageHeight = `${(29 / count) - ((count - 1)/count)}rem`;
-
-        return (
-            <div className="secondary-images">
-                {images.map(image => {
-                    return this.renderSecondaryImage(image, imageHeight)
-                })}
-                {this.renderShowAll(extraImages, imageHeight)}
-            </div>
-        )
-    }
-
-    renderSecondaryImage(image, imageHeight) {
+    renderCarousel() {
+        const { selectedImageIndex, animating, animationIndex } = this.state;
+        const { images } = this.props;
+        if (!images || !images.length) return;
         return (
-            <div key={image.id} className="image-wrapper">
-                <div className="image"
-                     style={{
-                         backgroundImage: `url(${image.url})`,
-                         height: imageHeight
-                     }}/>
-                <div className="overlay" onClick={() => this.openModal(image)} style={{height: imageHeight}}>
-                    <i className="material-icons">
-                        open_in_new
-                    </i>
-                </div>
+            <div className="image-selection">
+                {
+                    images.map((image, index) => {
+                        let className = 'image-option';
+                        if ((animating && index === animationIndex) || (!animating && index === selectedImageIndex)) {
+                            className += ' selected';
+                        }
+                        if (animating) className += ' animating';
+                        return (
+                            <span className={className}
+                                  onClick={() => {
+                                      if (this.state.animating) return;
+                                      this.selectImage(index);
+                                  }}/>
+                        )
+                    })
+                }
             </div>
-        )
+        );
```

Comment:
Is it normal to have semi-colons after after `return` statements? I think we're inconsistent with that.

Reply:
Is it normal to have semi-colons after `return` statements in JS: Yes
Is it normal to have semi-colons after `return` statements in React that return JSX (The JS/HTML stuff wrapped in `()`): I'm not sure

But since the first one is true, I tend to do that when I remember although it's easy to forget with these massive blocks of JSX in between sometimes